File: app/src/main/java/br/pedroso/upcomingmovies/moviedetails/di/MovieDetailsComponent.java
package br.pedroso.upcomingmovies.moviedetails.di;

import br.pedroso.upcomingmovies.di.ApplicationComponent;
import br.pedroso.upcomingmovies.di.scopes.FragmentScope;
import br.pedroso.upcomingmovies.moviedetails.MovieDetailsActivity;
import dagger.Component;

@FragmentScope
@Component(dependencies = ApplicationComponent.class, modules = MovieDetailsPresenterModule.class)
public interface MovieDetailsComponent {
    void inject(MovieDetailsActivity activity);
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/moviedetails/di/MovieDetailsComponent.kt
package br.pedroso.upcomingmovies.moviedetails.di

import br.pedroso.upcomingmovies.di.ApplicationComponent
import br.pedroso.upcomingmovies.di.scopes.FragmentScope
import br.pedroso.upcomingmovies.moviedetails.MovieDetailsActivity
import dagger.Component

@FragmentScope
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [MovieDetailsModule::class]
)
interface MovieDetailsComponent {
    fun inject(activity: MovieDetailsActivity)
}
